### PR TITLE
V2 poll results Part 2

### PR DESCRIPTION
**Pull request by norkans7**
This will use the backfilled results on the public site

### CODE REVIEW

**Comment by norkans7** [comment]
@nicpottier please review

File: templates/polls/poll_questions.haml [hunk 1]
```diff
@@ -13,6 +13,20 @@
...
+    %form.smartmin-form.form-horizontal{action:"{% url 'polls.poll_pull_refresh' object.id %}", method:"post",  enctype:"multipart/form-data"}
+      - csrf_token
+      .control-group
+        %label.control-label
+          Poll Results Data
+        .controls
+          .row
+            .col-md-3
+              %button.btn.btn-primary.btn-block{type:"submit", name:"poll", value:"{{object.id}}"}
+                Pull Refresh Results
```
Comment (nicpottier) [inline templates/polls/poll_questions.haml]: `Refresh Results`

File: ureport/polls/models.py [hunk 2]
```diff
@@ -510,44 +526,126 @@ def fetch_results(self, segment=None):
             traceback.print_exc()
 
     def get_results(self, segment=None):
```
Comment (nicpottier) [inline ureport/polls/models.py]: This is called for every page question on every page hit?

Comment (norkans7) [inline ureport/polls/models.py]: yes

Comment (norkans7) [inline ureport/polls/models.py]: may be we can add a five minute cache

Comment (nicpottier) [inline ureport/polls/models.py]: Ya, how expensive is that crazy query against a bit nigeria poll?

Comment (norkans7) [inline ureport/polls/models.py]: I'll check that too

File: ureport/polls/models.py [hunk 3]
```diff
@@ -550,8 +550,8 @@ def get_results(self, segment=None):
             cursor = connection.cursor()
 
             custom_sql = """
-                      SELECT w.label, count(*) AS count FROM (SELECT regexp_split_to_table(LOWER(text), E'[^[:alnum:]_]') AS label FROM polls_pollresult WHERE polls_pollresult.ruleset = '%s') w group by w.label order by count desc;
-                      """ % self.ruleset_uuid
+                      SELECT w.label, count(*) AS count FROM (SELECT regexp_split_to_table(LOWER(text), E'[^[:alnum:]_]') AS label FROM polls_pollresult WHERE polls_pollresult.org_id = %d AND polls_pollresult.flow = '%s' AND polls_pollresult.ruleset = '%s') w group by w.label order by count desc;
```
Comment (norkans7) [inline ureport/polls/models.py]: @nicpottier this change to use the existing index on org and flow 
and it improve this query time form 42000ms to 500ms

#150 will improve more to 120ms

**Comment by nicpottier** [comment]
goforit

**Pull request merged**